### DISCUSSION TITLE
Reading max17043 on ESP32S2 Feather

### DISCUSSION THREAD

**Discussion by dstnceswmer** [micropython/micropython#12513] [ESP32]
Hi I am working with the ESP32-S2 Feather - 2 MB PSRAM and Stemma QT/Qwiic Adafruit 5000 that I picked up from here:
https://www.amazon.com/dp/B09MBLN5V7?psc=1&ref=ppx_yo2ov_dt_b_product_details

I am trying to read from the onboard max17043 LiPO monitoring device to see the battery level. The Adafruit site for this product says it should be available at address 0x36:
https://learn.adafruit.com/adafruit-esp32-s2-feather/power-management

The problem is any libraries/examples I've been able to find are all Circuit Python not Micropython. I have connected t external devices over I2C fine so far but am unsure how to connect to this on-board device without knowing the SCL/SDA pin numbers etc...
Searching for a schematic of this board has not been of much help either.

Looking for any assistance on getting this working. Let me know if there's any more information I should provide.

Thanks in advance!

**Comment by robert-hh** [comment]
The schematics seem to be here: https://learn.adafruit.com/adafruit-esp32-s2-feather/downloads
SCA at GPIO3, SCL at GPIO4.

**Reply by dstnceswmer** [reply]
Ah, thanks! Unfortunately it still does not seem to be detecting any devices :/
```
from machine import I2C, Pin
>>> i2c = I2C(1, scl=Pin(4), sda=Pin(3))
>>> i2c.scan()
[]
>>> i2c = I2C(0, scl=Pin(4), sda=Pin(3))
>>> i2c.scan()
[]
>>> 
```

**Comment by robert-hh** [comment]
There is GPIO7 called I2C_PWR, and the I2C pull-up resistors are connected to VSENSOR. You could try to set Pin(7) to 1.
You could check the level at SDA and SCL. It should be almost 3.3V.

**Reply by dstnceswmer** [reply]
Ok, I think were getting somewhere.
Setting pin 7 high and now seeing a device at 54 (0x36) which should be the max17043 device!
```
>>> pin = Pin(7, Pin.OUT)
>>> pin.value(1)
>>> i2c.scan()
[54]
```

**Reply by dstnceswmer** [reply]
I did find someone's attempt at a micropython library for the max device here:
https://github.com/andrethemac/max17043.py

After manually setting pin7 to high, and then using pins 3/4 for sda/scl I was able to run the getVCell method and return around 3.3v.
What I'm not understanding then I guess is how to read the actual battery voltage (which is showing around 4.1v on my DMM) rather than the VSENSOR voltage which I'm assuming is the regulated voltage powering the ESP32.

**Reply by dstnceswmer** [reply]
The SoC (State of Charge) value does seem to be going down if battery is left on for a while without being plugged in, and increases when plugged in, indicating it is reflecting the charging/discharging. I just still do not understand why the reported cell voltage is around 3.3V phyisical measurement of the battery show it at around 4.1v-4.2v when fully charged.

**Reply by 2dof** [reply]
How did You make the battery voltage measurement with multimeter ? Was it with connected max17043 to battery and on  max17043  Cell input  , or just only on discontented battery (without load) ?
